diff --git a/BlitzDiffApp/app/hooks/useCurrentUser.ts b/BlitzDiffApp/app/hooks/useCurrentUser.ts
index 2938468c7..47ebaf5ee 100644
--- a/BlitzDiffApp/app/hooks/useCurrentUser.ts
+++ b/BlitzDiffApp/app/hooks/useCurrentUser.ts
@@ -1,7 +1,10 @@
-import { useQuery } from "blitz"
+import { useQuery, useSession } from "blitz"
 import getCurrentUser from "app/users/queries/getCurrentUser"
 
 export const useCurrentUser = () => {
-  const [user] = useQuery(getCurrentUser, null)
-  return user
+  // We wouldn't have to useSession() here, but doing so improves perf on initial
+  // load since we can skip the getCurrentUser() request.
+  const session = useSession()
+  const [user] = useQuery(getCurrentUser, null, { enabled: !!session.userId })
+  return session.userId ? user : null
 }
diff --git a/BlitzDiffApp/package.json b/BlitzDiffApp/package.json
index fa6a8f436..f659cf960 100644
--- a/BlitzDiffApp/package.json
+++ b/BlitzDiffApp/package.json
@@ -33,7 +33,7 @@
   "dependencies": {
     "@prisma/cli": "2.18.0",
     "@prisma/client": "2.18.0",
-    "blitz": "0.31.1",
+    "blitz": "0.30.0-canary.15",
     "react": "0.0.0-experimental-4ead6b530",
     "react-dom": "0.0.0-experimental-4ead6b530",
     "react-error-boundary": "2.3.2",
