diff --git a/BlitzDiffApp/.prettierignore b/BlitzDiffApp/.prettierignore
index ad8c486b6..c61fdf72d 100644
--- a/BlitzDiffApp/.prettierignore
+++ b/BlitzDiffApp/.prettierignore
@@ -3,5 +3,3 @@
 *.ico
 *.lock
 db/migrations
-.next
-.blitz
diff --git a/BlitzDiffApp/app/auth/auth-utils.ts b/BlitzDiffApp/app/auth/auth-utils.ts
index f5a851738..90db006d1 100644
--- a/BlitzDiffApp/app/auth/auth-utils.ts
+++ b/BlitzDiffApp/app/auth/auth-utils.ts
@@ -2,15 +2,15 @@ import { AuthenticationError } from "blitz"
 import SecurePassword from "secure-password"
 import db from "db"
 
-const SP = () => new SecurePassword()
+const SP = new SecurePassword()
 
 export const hashPassword = async (password: string) => {
-  const hashedBuffer = await SP().hash(Buffer.from(password))
+  const hashedBuffer = await SP.hash(Buffer.from(password))
   return hashedBuffer.toString("base64")
 }
 export const verifyPassword = async (hashedPassword: string, password: string) => {
   try {
-    return await SP().verify(Buffer.from(password), Buffer.from(hashedPassword, "base64"))
+    return await SP.verify(Buffer.from(password), Buffer.from(hashedPassword, "base64"))
   } catch (error) {
     console.error(error)
     return false
@@ -18,7 +18,7 @@ export const verifyPassword = async (hashedPassword: string, password: string) =
 }
 
 export const authenticateUser = async (email: string, password: string) => {
-  const user = await db.user.findFirst({ where: { email: email.toLowerCase() } })
+  const user = await db.user.findOne({ where: { email: email.toLowerCase() } })
 
   if (!user || !user.hashedPassword) throw new AuthenticationError()
 
diff --git a/BlitzDiffApp/app/hooks/useCurrentUser.ts b/BlitzDiffApp/app/hooks/useCurrentUser.ts
index 2938468c7..47ebaf5ee 100644
--- a/BlitzDiffApp/app/hooks/useCurrentUser.ts
+++ b/BlitzDiffApp/app/hooks/useCurrentUser.ts
@@ -1,7 +1,10 @@
-import { useQuery } from "blitz"
+import { useQuery, useSession } from "blitz"
 import getCurrentUser from "app/users/queries/getCurrentUser"
 
 export const useCurrentUser = () => {
-  const [user] = useQuery(getCurrentUser, null)
-  return user
+  // We wouldn't have to useSession() here, but doing so improves perf on initial
+  // load since we can skip the getCurrentUser() request.
+  const session = useSession()
+  const [user] = useQuery(getCurrentUser, null, { enabled: !!session.userId })
+  return session.userId ? user : null
 }
diff --git a/BlitzDiffApp/app/users/queries/getCurrentUser.ts b/BlitzDiffApp/app/users/queries/getCurrentUser.ts
index 9822e16bf..da5748deb 100644
--- a/BlitzDiffApp/app/users/queries/getCurrentUser.ts
+++ b/BlitzDiffApp/app/users/queries/getCurrentUser.ts
@@ -4,7 +4,7 @@ import db from "db"
 export default async function getCurrentUser(_ = null, { session }: Ctx) {
   if (!session.userId) return null
 
-  const user = await db.user.findFirst({
+  const user = await db.user.findOne({
     where: { id: session.userId },
     select: { id: true, name: true, email: true, role: true },
   })
diff --git a/BlitzDiffApp/package.json b/BlitzDiffApp/package.json
index 6fda6458c..f659cf960 100644
--- a/BlitzDiffApp/package.json
+++ b/BlitzDiffApp/package.json
@@ -31,9 +31,9 @@
     ]
   },
   "dependencies": {
-    "@prisma/cli": "~2.12",
-    "@prisma/client": "~2.12",
-    "blitz": "0.31.1",
+    "@prisma/cli": "2.18.0",
+    "@prisma/client": "2.18.0",
+    "blitz": "0.30.0-canary.15",
     "react": "0.0.0-experimental-4ead6b530",
     "react-dom": "0.0.0-experimental-4ead6b530",
     "react-error-boundary": "2.3.2",
