diff --git a/BlitzDiffApp/.gitignore b/BlitzDiffApp/.gitignore
index 3665bf5ce..70053f9b5 100644
--- a/BlitzDiffApp/.gitignore
+++ b/BlitzDiffApp/.gitignore
@@ -11,6 +11,7 @@ web_modules/
 /.blitz/
 /.next/
 *.sqlite
+*.sqlite-journal
 .now
 .blitz-console-history
 blitz-log.log
diff --git a/BlitzDiffApp/.npmrc b/BlitzDiffApp/.npmrc
index cffe8cdef..1b78f1c6f 100644
--- a/BlitzDiffApp/.npmrc
+++ b/BlitzDiffApp/.npmrc
@@ -1 +1,2 @@
 save-exact=true
+legacy-peer-deps=true
diff --git a/BlitzDiffApp/README.md b/BlitzDiffApp/README.md
index f1ff724d4..6d00b24a4 100644
--- a/BlitzDiffApp/README.md
+++ b/BlitzDiffApp/README.md
@@ -9,7 +9,7 @@ This is a [Blitz.js](https://github.com/blitz-js/blitz) app.
 Run your app in the development mode.
 
 ```
-blitz start
+blitz dev
 ```
 
 Open [http://localhost:3000](http://localhost:3000) with your browser to see the result.
@@ -33,8 +33,6 @@ DATABASE_URL=postgresql://<YOUR_DB_USERNAME>@localhost:5432/BlitzDiffApp_test
 Runs your tests using Jest.
 
 ```
-blitz test
-or
 yarn test
 ```
 
@@ -47,12 +45,13 @@ Blitz comes with a powerful CLI that is designed to make development easy and fa
 ```
   blitz [COMMAND]
 
+  dev       Start a development server
   build     Create a production build
-  console   Run the Blitz console REPL
-  db        Run database commands
+  start     Start a production server
+  prisma    Run prisma commands
   generate  Generate new files for your Blitz project
+  console   Run the Blitz console REPL
   help      display help for blitz
-  start     Start a development server
   test      Run project tests
 ```
 
@@ -65,20 +64,6 @@ Here is the starting structure of your app.
 ```
 BlitzDiffApp
 ├── app/
-│   ├── core/
-│   │   ├── components/
-│   │   │   ├── Form.tsx
-│   │   │   └── LabeledTextField.tsx
-│   │   ├── hooks/
-│   │   │   └── useCurrentUser.ts
-│   │   └── layouts/
-│   │       └── Layout.tsx
-│   ├── pages/
-│   │   ├── 404.tsx
-│   │   ├── _app.tsx
-│   │   ├── _document.tsx
-│   │   ├── index.test.tsx
-│   │   └── index.tsx
 │   ├── api/
 │   ├── auth/
 │   │   ├── components/
@@ -86,13 +71,33 @@ BlitzDiffApp
 │   │   │   └── SignupForm.tsx
 │   │   ├── mutations/
 │   │   │   ├── changePassword.ts
+│   │   │   ├── forgotPassword.test.ts
+│   │   │   ├── forgotPassword.ts
 │   │   │   ├── login.ts
 │   │   │   ├── logout.ts
+│   │   │   ├── resetPassword.test.ts
+│   │   │   ├── resetPassword.ts
 │   │   │   └── signup.ts
 │   │   ├── pages/
+│   │   │   ├── forgot-password.tsx
 │   │   │   ├── login.tsx
+│   │   │   ├── reset-password.tsx
 │   │   │   └── signup.tsx
 │   │   └── validations.ts
+│   ├── core/
+│   │   ├── components/
+│   │   │   ├── Form.tsx
+│   │   │   └── LabeledTextField.tsx
+│   │   ├── hooks/
+│   │   │   └── useCurrentUser.ts
+│   │   └── layouts/
+│   │       └── Layout.tsx
+│   ├── pages/
+│   │   ├── 404.tsx
+│   │   ├── _app.tsx
+│   │   ├── _document.tsx
+│   │   ├── index.test.tsx
+│   │   └── index.tsx
 │   └── users/
 │       └── queries/
 │           └── getCurrentUser.ts
@@ -101,6 +106,8 @@ BlitzDiffApp
 │   ├── schema.prisma
 │   └── seeds.ts
 ├── integrations/
+├── mailers/
+│   └── forgotPasswordMailer.ts
 ├── public/
 │   ├── favicon.ico*
 │   └── logo.png
@@ -113,6 +120,7 @@ BlitzDiffApp
 ├── jest.config.js
 ├── package.json
 ├── tsconfig.json
+├── types.d.ts
 ├── types.ts
 └── yarn.lock
 ```
diff --git a/BlitzDiffApp/app/auth/components/LoginForm.tsx b/BlitzDiffApp/app/auth/components/LoginForm.tsx
index e23765149..8861cf009 100644
--- a/BlitzDiffApp/app/auth/components/LoginForm.tsx
+++ b/BlitzDiffApp/app/auth/components/LoginForm.tsx
@@ -1,9 +1,8 @@
-import React from "react"
 import { AuthenticationError, Link, useMutation } from "blitz"
 import { LabeledTextField } from "app/core/components/LabeledTextField"
 import { Form, FORM_ERROR } from "app/core/components/Form"
 import login from "app/auth/mutations/login"
-import { LoginInput } from "app/auth/validations"
+import { Login } from "app/auth/validations"
 
 type LoginFormProps = {
   onSuccess?: () => void
@@ -18,7 +17,7 @@ export const LoginForm = (props: LoginFormProps) => {
 
       <Form
         submitText="Login"
-        schema={LoginInput}
+        schema={Login}
         initialValues={{ email: "", password: "" }}
         onSubmit={async (values) => {
           try {
diff --git a/BlitzDiffApp/app/auth/components/SignupForm.tsx b/BlitzDiffApp/app/auth/components/SignupForm.tsx
index e6315e842..7362e35d0 100644
--- a/BlitzDiffApp/app/auth/components/SignupForm.tsx
+++ b/BlitzDiffApp/app/auth/components/SignupForm.tsx
@@ -1,9 +1,8 @@
-import React from "react"
 import { useMutation } from "blitz"
 import { LabeledTextField } from "app/core/components/LabeledTextField"
 import { Form, FORM_ERROR } from "app/core/components/Form"
 import signup from "app/auth/mutations/signup"
-import { SignupInput } from "app/auth/validations"
+import { Signup } from "app/auth/validations"
 
 type SignupFormProps = {
   onSuccess?: () => void
@@ -18,7 +17,7 @@ export const SignupForm = (props: SignupFormProps) => {
 
       <Form
         submitText="Create Account"
-        schema={SignupInput}
+        schema={Signup}
         initialValues={{ email: "", password: "" }}
         onSubmit={async (values) => {
           try {
diff --git a/BlitzDiffApp/app/auth/mutations/changePassword.ts b/BlitzDiffApp/app/auth/mutations/changePassword.ts
index 64997cbd3..5cc49a9d7 100644
--- a/BlitzDiffApp/app/auth/mutations/changePassword.ts
+++ b/BlitzDiffApp/app/auth/mutations/changePassword.ts
@@ -1,21 +1,23 @@
-import { Ctx, NotFoundError, SecurePassword } from "blitz"
+import { NotFoundError, SecurePassword, resolver } from "blitz"
 import db from "db"
 import { authenticateUser } from "./login"
-import { ChangePasswordInput, ChangePasswordInputType } from "../validations"
+import { ChangePassword } from "../validations"
 
-export default async function changePassword(input: ChangePasswordInputType, ctx: Ctx) {
-  ctx.session.authorize()
-  const { currentPassword, newPassword } = ChangePasswordInput.parse(input)
+export default resolver.pipe(
+  resolver.zod(ChangePassword),
+  resolver.authorize(),
+  async ({ currentPassword, newPassword }, ctx) => {
+    const user = await db.user.findFirst({ where: { id: ctx.session.userId! } })
+    if (!user) throw new NotFoundError()
 
-  const user = await db.user.findFirst({ where: { id: ctx.session.userId } })
-  if (!user) throw new NotFoundError()
-  await authenticateUser(user.email, currentPassword)
+    await authenticateUser(user.email, currentPassword)
 
-  const hashedPassword = await SecurePassword.hash(newPassword)
-  await db.user.update({
-    where: { id: user.id },
-    data: { hashedPassword },
-  })
+    const hashedPassword = await SecurePassword.hash(newPassword)
+    await db.user.update({
+      where: { id: user.id },
+      data: { hashedPassword },
+    })
 
-  return true
-}
+    return true
+  }
+)
diff --git a/BlitzDiffApp/app/auth/mutations/forgotPassword.test.ts b/BlitzDiffApp/app/auth/mutations/forgotPassword.test.ts
index 1b3f47d84..d5d0098f9 100644
--- a/BlitzDiffApp/app/auth/mutations/forgotPassword.test.ts
+++ b/BlitzDiffApp/app/auth/mutations/forgotPassword.test.ts
@@ -1,4 +1,4 @@
-import { hash256 } from "blitz"
+import { hash256, Ctx } from "blitz"
 import forgotPassword from "./forgotPassword"
 import db from "db"
 import previewEmail from "preview-email"
@@ -16,7 +16,7 @@ jest.mock("preview-email", () => jest.fn())
 
 describe("forgotPassword mutation", () => {
   it("does not throw error if user doesn't exist", async () => {
-    await expect(forgotPassword({ email: "no-user@email.com" })).resolves.not.toThrow()
+    await expect(forgotPassword({ email: "no-user@email.com" }, {} as Ctx)).resolves.not.toThrow()
   })
 
   it("works correctly", async () => {
@@ -38,7 +38,7 @@ describe("forgotPassword mutation", () => {
     })
 
     // Invoke the mutation
-    await forgotPassword({ email: user.email })
+    await forgotPassword({ email: user.email }, {} as Ctx)
 
     const tokens = await db.token.findMany({ where: { userId: user.id } })
     const token = tokens[0]
diff --git a/BlitzDiffApp/app/auth/mutations/forgotPassword.ts b/BlitzDiffApp/app/auth/mutations/forgotPassword.ts
index 08125c2a2..1cbf2a33c 100644
--- a/BlitzDiffApp/app/auth/mutations/forgotPassword.ts
+++ b/BlitzDiffApp/app/auth/mutations/forgotPassword.ts
@@ -1,13 +1,11 @@
-import { generateToken, hash256 } from "blitz"
+import { resolver, generateToken, hash256 } from "blitz"
 import db from "db"
 import { forgotPasswordMailer } from "mailers/forgotPasswordMailer"
-import { ForgotPasswordInput, ForgotPasswordInputType } from "../validations"
+import { ForgotPassword } from "../validations"
 
 const RESET_PASSWORD_TOKEN_EXPIRATION_IN_HOURS = 4
 
-export default async function forgotPassword(input: ForgotPasswordInputType) {
-  const { email } = ForgotPasswordInput.parse(input)
-
+export default resolver.pipe(resolver.zod(ForgotPassword), async ({ email }) => {
   // 1. Get the user
   const user = await db.user.findFirst({ where: { email: email.toLowerCase() } })
 
@@ -40,4 +38,4 @@ export default async function forgotPassword(input: ForgotPasswordInputType) {
 
   // 8. Return the same result whether a password reset email was sent or not
   return
-}
+})
diff --git a/BlitzDiffApp/app/auth/mutations/login.ts b/BlitzDiffApp/app/auth/mutations/login.ts
index 3b010e651..b155223a2 100644
--- a/BlitzDiffApp/app/auth/mutations/login.ts
+++ b/BlitzDiffApp/app/auth/mutations/login.ts
@@ -1,6 +1,7 @@
-import { Ctx, SecurePassword, AuthenticationError } from "blitz"
+import { resolver, SecurePassword, AuthenticationError } from "blitz"
 import db from "db"
-import { LoginInput, LoginInputType } from "../validations"
+import { Login } from "../validations"
+import { Role } from "types"
 
 export const authenticateUser = async (email: string, password: string) => {
   const user = await db.user.findFirst({ where: { email } })
@@ -18,14 +19,11 @@ export const authenticateUser = async (email: string, password: string) => {
   return rest
 }
 
-export default async function login(input: LoginInputType, ctx: Ctx) {
-  // This throws an error if input is invalid
-  const { email, password } = LoginInput.parse(input)
-
+export default resolver.pipe(resolver.zod(Login), async ({ email, password }, ctx) => {
   // This throws an error if credentials are invalid
   const user = await authenticateUser(email, password)
 
-  await ctx.session.create({ userId: user.id, roles: [user.role] })
+  await ctx.session.$create({ userId: user.id, role: user.role as Role })
 
   return user
-}
+})
diff --git a/BlitzDiffApp/app/auth/mutations/logout.ts b/BlitzDiffApp/app/auth/mutations/logout.ts
index eeaca86d1..c2f2fefd8 100644
--- a/BlitzDiffApp/app/auth/mutations/logout.ts
+++ b/BlitzDiffApp/app/auth/mutations/logout.ts
@@ -1,5 +1,5 @@
 import { Ctx } from "blitz"
 
 export default async function logout(_: any, ctx: Ctx) {
-  return await ctx.session.revoke()
+  return await ctx.session.$revoke()
 }
diff --git a/BlitzDiffApp/app/auth/mutations/resetPassword.ts b/BlitzDiffApp/app/auth/mutations/resetPassword.ts
index 0bf78cb4d..ece656d4c 100644
--- a/BlitzDiffApp/app/auth/mutations/resetPassword.ts
+++ b/BlitzDiffApp/app/auth/mutations/resetPassword.ts
@@ -1,6 +1,6 @@
-import { Ctx, SecurePassword, hash256 } from "blitz"
+import { resolver, SecurePassword, hash256 } from "blitz"
 import db from "db"
-import { ResetPasswordInput, ResetPasswordInputType } from "../validations"
+import { ResetPassword } from "../validations"
 import login from "./login"
 
 export class ResetPasswordError extends Error {
@@ -8,9 +8,7 @@ export class ResetPasswordError extends Error {
   message = "Reset password link is invalid or it has expired."
 }
 
-export default async function resetPassword(input: ResetPasswordInputType, ctx: Ctx) {
-  const { password, token } = ResetPasswordInput.parse(input)
-
+export default resolver.pipe(resolver.zod(ResetPassword), async ({ password, token }, ctx) => {
   // 1. Try to find this token in the database
   const hashedToken = hash256(token)
   const possibleToken = await db.token.findFirst({
@@ -46,4 +44,4 @@ export default async function resetPassword(input: ResetPasswordInputType, ctx:
   await login({ email: user.email, password }, ctx)
 
   return true
-}
+})
diff --git a/BlitzDiffApp/app/auth/mutations/signup.ts b/BlitzDiffApp/app/auth/mutations/signup.ts
index f10cf75e7..70d025ed6 100644
--- a/BlitzDiffApp/app/auth/mutations/signup.ts
+++ b/BlitzDiffApp/app/auth/mutations/signup.ts
@@ -1,18 +1,15 @@
-import { Ctx, SecurePassword } from "blitz"
+import { resolver, SecurePassword } from "blitz"
 import db from "db"
-import { SignupInput, SignupInputType } from "app/auth/validations"
-
-export default async function signup(input: SignupInputType, ctx: Ctx) {
-  // This throws an error if input is invalid
-  const { email, password } = SignupInput.parse(input)
+import { Signup } from "app/auth/validations"
+import { Role } from "types"
 
+export default resolver.pipe(resolver.zod(Signup), async ({ email, password }, ctx) => {
   const hashedPassword = await SecurePassword.hash(password)
   const user = await db.user.create({
-    data: { email: email.toLowerCase(), hashedPassword, role: "user" },
+    data: { email: email.toLowerCase(), hashedPassword, role: "USER" },
     select: { id: true, name: true, email: true, role: true },
   })
 
-  await ctx.session.create({ userId: user.id, roles: [user.role] })
-
+  await ctx.session.$create({ userId: user.id, role: user.role as Role })
   return user
-}
+})
diff --git a/BlitzDiffApp/app/auth/pages/forgot-password.tsx b/BlitzDiffApp/app/auth/pages/forgot-password.tsx
index 2cac6b9f2..0d9a295ce 100644
--- a/BlitzDiffApp/app/auth/pages/forgot-password.tsx
+++ b/BlitzDiffApp/app/auth/pages/forgot-password.tsx
@@ -2,7 +2,7 @@ import { BlitzPage, useMutation } from "blitz"
 import Layout from "app/core/layouts/Layout"
 import { LabeledTextField } from "app/core/components/LabeledTextField"
 import { Form, FORM_ERROR } from "app/core/components/Form"
-import { ForgotPasswordInput } from "app/auth/validations"
+import { ForgotPassword } from "app/auth/validations"
 import forgotPassword from "app/auth/mutations/forgotPassword"
 
 const ForgotPasswordPage: BlitzPage = () => {
@@ -23,7 +23,7 @@ const ForgotPasswordPage: BlitzPage = () => {
       ) : (
         <Form
           submitText="Send Reset Password Instructions"
-          schema={ForgotPasswordInput}
+          schema={ForgotPassword}
           initialValues={{ email: "" }}
           onSubmit={async (values) => {
             try {
@@ -42,6 +42,7 @@ const ForgotPasswordPage: BlitzPage = () => {
   )
 }
 
+ForgotPasswordPage.redirectAuthenticatedTo = "/"
 ForgotPasswordPage.getLayout = (page) => <Layout title="Forgot Your Password?">{page}</Layout>
 
 export default ForgotPasswordPage
diff --git a/BlitzDiffApp/app/auth/pages/login.tsx b/BlitzDiffApp/app/auth/pages/login.tsx
index 6e18bd89a..45e76ff56 100644
--- a/BlitzDiffApp/app/auth/pages/login.tsx
+++ b/BlitzDiffApp/app/auth/pages/login.tsx
@@ -1,4 +1,3 @@
-import React from "react"
 import { useRouter, BlitzPage } from "blitz"
 import Layout from "app/core/layouts/Layout"
 import { LoginForm } from "app/auth/components/LoginForm"
@@ -8,11 +7,17 @@ const LoginPage: BlitzPage = () => {
 
   return (
     <div>
-      <LoginForm onSuccess={() => router.push("/")} />
+      <LoginForm
+        onSuccess={() => {
+          const next = (router.query.next as string) ?? "/"
+          router.push(next)
+        }}
+      />
     </div>
   )
 }
 
+LoginPage.redirectAuthenticatedTo = "/"
 LoginPage.getLayout = (page) => <Layout title="Log In">{page}</Layout>
 
 export default LoginPage
diff --git a/BlitzDiffApp/app/auth/pages/reset-password.tsx b/BlitzDiffApp/app/auth/pages/reset-password.tsx
index 8068a11e9..405d53814 100644
--- a/BlitzDiffApp/app/auth/pages/reset-password.tsx
+++ b/BlitzDiffApp/app/auth/pages/reset-password.tsx
@@ -2,7 +2,7 @@ import { BlitzPage, useRouterQuery, Link, useMutation } from "blitz"
 import Layout from "app/core/layouts/Layout"
 import { LabeledTextField } from "app/core/components/LabeledTextField"
 import { Form, FORM_ERROR } from "app/core/components/Form"
-import { ResetPasswordInput } from "app/auth/validations"
+import { ResetPassword } from "app/auth/validations"
 import resetPassword from "app/auth/mutations/resetPassword"
 
 const ResetPasswordPage: BlitzPage = () => {
@@ -23,11 +23,11 @@ const ResetPasswordPage: BlitzPage = () => {
       ) : (
         <Form
           submitText="Reset Password"
-          schema={ResetPasswordInput}
-          initialValues={{ password: "", passwordConfirmation: "", token: query.token as string }}
+          schema={ResetPassword.omit({ token: true })}
+          initialValues={{ password: "", passwordConfirmation: "" }}
           onSubmit={async (values) => {
             try {
-              await resetPasswordMutation(values)
+              await resetPasswordMutation({ ...values, token: query.token as string })
             } catch (error) {
               if (error.name === "ResetPasswordError") {
                 return {
@@ -53,6 +53,7 @@ const ResetPasswordPage: BlitzPage = () => {
   )
 }
 
+ResetPasswordPage.redirectAuthenticatedTo = "/"
 ResetPasswordPage.getLayout = (page) => <Layout title="Reset Your Password">{page}</Layout>
 
 export default ResetPasswordPage
diff --git a/BlitzDiffApp/app/auth/pages/signup.tsx b/BlitzDiffApp/app/auth/pages/signup.tsx
index 1ef9c2be2..ea9de9cd1 100644
--- a/BlitzDiffApp/app/auth/pages/signup.tsx
+++ b/BlitzDiffApp/app/auth/pages/signup.tsx
@@ -1,4 +1,3 @@
-import React from "react"
 import { useRouter, BlitzPage } from "blitz"
 import Layout from "app/core/layouts/Layout"
 import { SignupForm } from "app/auth/components/SignupForm"
@@ -13,6 +12,7 @@ const SignupPage: BlitzPage = () => {
   )
 }
 
+SignupPage.redirectAuthenticatedTo = "/"
 SignupPage.getLayout = (page) => <Layout title="Sign Up">{page}</Layout>
 
 export default SignupPage
diff --git a/BlitzDiffApp/app/auth/validations.ts b/BlitzDiffApp/app/auth/validations.ts
index 2c4aaf522..6f4378992 100644
--- a/BlitzDiffApp/app/auth/validations.ts
+++ b/BlitzDiffApp/app/auth/validations.ts
@@ -2,24 +2,21 @@ import * as z from "zod"
 
 const password = z.string().min(10).max(100)
 
-export const SignupInput = z.object({
+export const Signup = z.object({
   email: z.string().email(),
   password,
 })
-export type SignupInputType = z.infer<typeof SignupInput>
 
-export const LoginInput = z.object({
+export const Login = z.object({
   email: z.string().email(),
   password: z.string(),
 })
-export type LoginInputType = z.infer<typeof LoginInput>
 
-export const ForgotPasswordInput = z.object({
+export const ForgotPassword = z.object({
   email: z.string().email(),
 })
-export type ForgotPasswordInputType = z.infer<typeof ForgotPasswordInput>
 
-export const ResetPasswordInput = z
+export const ResetPassword = z
   .object({
     password: password,
     passwordConfirmation: password,
@@ -29,10 +26,8 @@ export const ResetPasswordInput = z
     message: "Passwords don't match",
     path: ["passwordConfirmation"], // set the path of the error
   })
-export type ResetPasswordInputType = z.infer<typeof ResetPasswordInput>
 
-export const ChangePasswordInput = z.object({
+export const ChangePassword = z.object({
   currentPassword: z.string(),
   newPassword: password,
 })
-export type ChangePasswordInputType = z.infer<typeof ChangePasswordInput>
diff --git a/BlitzDiffApp/app/core/components/Form.tsx b/BlitzDiffApp/app/core/components/Form.tsx
index 1e5e0f0c3..2951abf33 100644
--- a/BlitzDiffApp/app/core/components/Form.tsx
+++ b/BlitzDiffApp/app/core/components/Form.tsx
@@ -1,17 +1,18 @@
-import React, { ReactNode, PropsWithoutRef } from "react"
+import { ReactNode, PropsWithoutRef } from "react"
 import { Form as FinalForm, FormProps as FinalFormProps } from "react-final-form"
 import * as z from "zod"
 export { FORM_ERROR } from "final-form"
 
-type FormProps<S extends z.ZodType<any, any>> = {
+export interface FormProps<S extends z.ZodType<any, any>>
+  extends Omit<PropsWithoutRef<JSX.IntrinsicElements["form"]>, "onSubmit"> {
   /** All your form fields */
-  children: ReactNode
+  children?: ReactNode
   /** Text to display in the submit button */
   submitText?: string
   schema?: S
   onSubmit: FinalFormProps<z.infer<S>>["onSubmit"]
   initialValues?: FinalFormProps<z.infer<S>>["initialValues"]
-} & Omit<PropsWithoutRef<JSX.IntrinsicElements["form"]>, "onSubmit">
+}
 
 export function Form<S extends z.ZodType<any, any>>({
   children,
diff --git a/BlitzDiffApp/app/core/components/LabeledTextField.tsx b/BlitzDiffApp/app/core/components/LabeledTextField.tsx
index a1235ffec..0d8517b8c 100644
--- a/BlitzDiffApp/app/core/components/LabeledTextField.tsx
+++ b/BlitzDiffApp/app/core/components/LabeledTextField.tsx
@@ -1,5 +1,5 @@
-import React, {PropsWithoutRef} from "react"
-import {useField} from "react-final-form"
+import { forwardRef, PropsWithoutRef } from "react"
+import { useField } from "react-final-form"
 
 export interface LabeledTextFieldProps extends PropsWithoutRef<JSX.IntrinsicElements["input"]> {
   /** Field name. */
@@ -11,13 +11,13 @@ export interface LabeledTextFieldProps extends PropsWithoutRef<JSX.IntrinsicElem
   outerProps?: PropsWithoutRef<JSX.IntrinsicElements["div"]>
 }
 
-export const LabeledTextField = React.forwardRef<HTMLInputElement, LabeledTextFieldProps>(
-  ({name, label, outerProps, ...props}, ref) => {
+export const LabeledTextField = forwardRef<HTMLInputElement, LabeledTextFieldProps>(
+  ({ name, label, outerProps, ...props }, ref) => {
     const {
       input,
-      meta: {touched, error, submitError, submitting},
+      meta: { touched, error, submitError, submitting },
     } = useField(name, {
-      parse: props.type === "number" ? Number : undefined
+      parse: props.type === "number" ? Number : undefined,
     })
 
     const normalizedError = Array.isArray(error) ? error.join(", ") : error || submitError
@@ -30,7 +30,7 @@ export const LabeledTextField = React.forwardRef<HTMLInputElement, LabeledTextFi
         </label>
 
         {touched && normalizedError && (
-          <div role="alert" style={{color: "red"}}>
+          <div role="alert" style={{ color: "red" }}>
             {normalizedError}
           </div>
         )}
@@ -53,7 +53,7 @@ export const LabeledTextField = React.forwardRef<HTMLInputElement, LabeledTextFi
         `}</style>
       </div>
     )
-  },
+  }
 )
 
 export default LabeledTextField
diff --git a/BlitzDiffApp/app/pages/_app.tsx b/BlitzDiffApp/app/pages/_app.tsx
index 465065a31..d6f7b831e 100644
--- a/BlitzDiffApp/app/pages/_app.tsx
+++ b/BlitzDiffApp/app/pages/_app.tsx
@@ -1,5 +1,12 @@
-import { AppProps, ErrorComponent, useRouter, AuthenticationError, AuthorizationError } from "blitz"
-import { ErrorBoundary, FallbackProps } from "react-error-boundary"
+import {
+  AppProps,
+  ErrorComponent,
+  useRouter,
+  AuthenticationError,
+  AuthorizationError,
+  ErrorFallbackProps,
+} from "blitz"
+import { ErrorBoundary } from "react-error-boundary"
 import { queryCache } from "react-query"
 import LoginForm from "app/auth/components/LoginForm"
 
@@ -22,22 +29,19 @@ export default function App({ Component, pageProps }: AppProps) {
   )
 }
 
-function RootErrorFallback({ error, resetErrorBoundary }: FallbackProps) {
+function RootErrorFallback({ error, resetErrorBoundary }: ErrorFallbackProps) {
   if (error instanceof AuthenticationError) {
     return <LoginForm onSuccess={resetErrorBoundary} />
   } else if (error instanceof AuthorizationError) {
     return (
       <ErrorComponent
-        statusCode={(error as any).statusCode}
+        statusCode={error.statusCode}
         title="Sorry, you are not authorized to access this"
       />
     )
   } else {
     return (
-      <ErrorComponent
-        statusCode={(error as any)?.statusCode || 400}
-        title={error?.message || error?.name}
-      />
+      <ErrorComponent statusCode={error.statusCode || 400} title={error.message || error.name} />
     )
   }
 }
diff --git a/BlitzDiffApp/app/pages/index.test.tsx b/BlitzDiffApp/app/pages/index.test.tsx
index 66aba7111..9374c6024 100644
--- a/BlitzDiffApp/app/pages/index.test.tsx
+++ b/BlitzDiffApp/app/pages/index.test.tsx
@@ -1,4 +1,3 @@
-import React from "react"
 import { render } from "test/utils"
 
 import Home from "./index"
diff --git a/BlitzDiffApp/app/pages/index.tsx b/BlitzDiffApp/app/pages/index.tsx
index c04bbb66c..087d51d96 100644
--- a/BlitzDiffApp/app/pages/index.tsx
+++ b/BlitzDiffApp/app/pages/index.tsx
@@ -73,9 +73,7 @@ const Home: BlitzPage = () => {
         <pre>
           <code>blitz generate all project name:string</code>
         </pre>
-        <pre>
-          <code>blitz prisma migrate dev --preview-feature</code>
-        </pre>
+        <div style={{ marginBottom: "1rem" }}>(And select Yes to run prisma migrate)</div>
         <div>
           <p>
             Then <strong>restart the server</strong>
@@ -84,7 +82,7 @@ const Home: BlitzPage = () => {
             <code>Ctrl + c</code>
           </pre>
           <pre>
-            <code>blitz start</code>
+            <code>blitz dev</code>
           </pre>
           <p>
             and go to{" "}
@@ -267,6 +265,7 @@ const Home: BlitzPage = () => {
   )
 }
 
+Home.suppressFirstRenderFlicker = true
 Home.getLayout = (page) => <Layout title="Home">{page}</Layout>
 
 export default Home
diff --git a/BlitzDiffApp/blitz.config.js b/BlitzDiffApp/blitz.config.js
index fa3795200..ce6217701 100644
--- a/BlitzDiffApp/blitz.config.js
+++ b/BlitzDiffApp/blitz.config.js
@@ -1,4 +1,4 @@
-const { sessionMiddleware, simpleRolesIsAuthorized } = require("@blitzjs/server")
+const { sessionMiddleware, simpleRolesIsAuthorized } = require("blitz")
 
 module.exports = {
   middleware: [
diff --git a/BlitzDiffApp/db/migrations/20210306011928_initial_migration/migration.sql b/BlitzDiffApp/db/migrations/20210306003910_initial_migration/migration.sql
similarity index 97%
rename from BlitzDiffApp/db/migrations/20210306011928_initial_migration/migration.sql
rename to BlitzDiffApp/db/migrations/20210306003910_initial_migration/migration.sql
index 6df839c7a..0b34cc718 100644
--- a/BlitzDiffApp/db/migrations/20210306011928_initial_migration/migration.sql
+++ b/BlitzDiffApp/db/migrations/20210306003910_initial_migration/migration.sql
@@ -6,7 +6,7 @@ CREATE TABLE "User" (
     "name" TEXT,
     "email" TEXT NOT NULL,
     "hashedPassword" TEXT,
-    "role" TEXT NOT NULL DEFAULT 'user'
+    "role" TEXT NOT NULL DEFAULT 'USER'
 );
 
 -- CreateTable
diff --git a/BlitzDiffApp/db/migrations/migration_lock.toml b/BlitzDiffApp/db/migrations/migration_lock.toml
index 963afcc9a..e5e5c4705 100644
--- a/BlitzDiffApp/db/migrations/migration_lock.toml
+++ b/BlitzDiffApp/db/migrations/migration_lock.toml
@@ -1,2 +1,3 @@
 # Please do not edit this file manually
+# It should be added in your version-control system (i.e. Git)
 provider = "sqlite"
\ No newline at end of file
diff --git a/BlitzDiffApp/db/schema.prisma b/BlitzDiffApp/db/schema.prisma
index dde411b62..1ad932610 100644
--- a/BlitzDiffApp/db/schema.prisma
+++ b/BlitzDiffApp/db/schema.prisma
@@ -13,20 +13,20 @@ generator client {
 // --------------------------------------
 
 model User {
-  id             Int       @default(autoincrement()) @id
-  createdAt      DateTime  @default(now())
-  updatedAt      DateTime  @updatedAt
+  id             Int      @id @default(autoincrement())
+  createdAt      DateTime @default(now())
+  updatedAt      DateTime @updatedAt
   name           String?
-  email          String    @unique
+  email          String   @unique
   hashedPassword String?
-  role           String    @default("user")
+  role           String   @default("USER")
 
-  tokens         Token[]
-  sessions       Session[]
+  tokens   Token[]
+  sessions Session[]
 }
 
 model Session {
-  id                 Int       @default(autoincrement()) @id
+  id                 Int       @id @default(autoincrement())
   createdAt          DateTime  @default(now())
   updatedAt          DateTime  @updatedAt
   expiresAt          DateTime?
@@ -36,14 +36,14 @@ model Session {
   publicData         String?
   privateData        String?
 
-  user               User?     @relation(fields: [userId], references: [id])
-  userId             Int?
+  user   User? @relation(fields: [userId], references: [id])
+  userId Int?
 }
 
 model Token {
-  id          Int       @default(autoincrement()) @id
-  createdAt   DateTime  @default(now())
-  updatedAt   DateTime  @updatedAt
+  id          Int      @id @default(autoincrement())
+  createdAt   DateTime @default(now())
+  updatedAt   DateTime @updatedAt
   hashedToken String
   type        String
   // See note below about TokenType enum
diff --git a/BlitzDiffApp/package.json b/BlitzDiffApp/package.json
index c00d2fbf5..2bf16e494 100644
--- a/BlitzDiffApp/package.json
+++ b/BlitzDiffApp/package.json
@@ -2,16 +2,14 @@
   "name": "BlitzDiffApp",
   "version": "1.0.0",
   "scripts": {
+    "dev": "blitz dev",
+    "build": "blitz build",
     "start": "blitz start",
     "studio": "blitz prisma studio",
-    "build": "blitz build",
     "lint": "eslint --ignore-path .gitignore --ext .js,.ts,.tsx .",
     "test": "jest",
     "test:watch": "jest --watch"
   },
-  "browserslist": [
-    "defaults"
-  ],
   "prisma": {
     "schema": "db/schema.prisma"
   },
@@ -31,13 +29,13 @@
     ]
   },
   "dependencies": {
-    "@prisma/cli": "~2.15",
-    "@prisma/client": "~2.15",
-    "blitz": "0.30.0-canary.15",
+    "prisma": "~2.17",
+    "@prisma/client": "~2.17",
+    "blitz": "0.31.1",
     "react": "0.0.0-experimental-3310209d0",
     "react-dom": "0.0.0-experimental-3310209d0",
     "react-error-boundary": "3.1.1",
-    "typescript": "4.2.3",
+    "typescript": "~4.1",
     "zod": "1.11.11",
     "final-form": "4.20.2",
     "react-final-form": "6.5.2"
@@ -50,7 +48,8 @@
     "lint-staged": "10.5.4",
     "prettier": "2.2.1",
     "pretty-quick": "3.1.0",
-    "preview-email": "3.0.3"
+    "preview-email": "3.0.3",
+    "prettier-plugin-prisma": "0.4.0"
   },
   "private": true
 }
diff --git a/BlitzDiffApp/test/setup.ts b/BlitzDiffApp/test/setup.ts
index 4ca742763..c278ddb9e 100644
--- a/BlitzDiffApp/test/setup.ts
+++ b/BlitzDiffApp/test/setup.ts
@@ -1,2 +1,4 @@
 // This is the jest 'setupFilesAfterEnv' setup file
 // It's a good place to set globals, add global before/after hooks, etc
+
+export {} // so TS doesn't complain
diff --git a/BlitzDiffApp/test/utils.tsx b/BlitzDiffApp/test/utils.tsx
index 856e68186..4bdf86ac3 100644
--- a/BlitzDiffApp/test/utils.tsx
+++ b/BlitzDiffApp/test/utils.tsx
@@ -1,4 +1,3 @@
-import React from "react"
 import { RouterContext, BlitzRouter } from "blitz"
 import { render as defaultRender } from "@testing-library/react"
 import { renderHook as defaultRenderHook } from "@testing-library/react-hooks"
@@ -71,6 +70,7 @@ export const mockRouter: BlitzRouter = {
   params: {},
   query: {},
   isReady: true,
+  isLocaleDomain: false,
   push: jest.fn(),
   replace: jest.fn(),
   reload: jest.fn(),
diff --git a/BlitzDiffApp/tsconfig.json b/BlitzDiffApp/tsconfig.json
index e1d2226fa..c6d5bee62 100644
--- a/BlitzDiffApp/tsconfig.json
+++ b/BlitzDiffApp/tsconfig.json
@@ -18,6 +18,6 @@
     "incremental": true,
     "tsBuildInfoFile": ".tsbuildinfo"
   },
-  "exclude": ["node_modules"],
+  "exclude": ["node_modules", "**/*.e2e.ts", "cypress"],
   "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"]
 }
diff --git a/BlitzDiffApp/types.d.ts b/BlitzDiffApp/types.d.ts
new file mode 100644
index 000000000..d35eda64a
--- /dev/null
+++ b/BlitzDiffApp/types.d.ts
@@ -0,0 +1,8 @@
+declare module "*.module.scss" {
+  const content: { [className: string]: string }
+  export default content
+}
+declare module "*.module.css" {
+  const content: { [className: string]: string }
+  export default content
+}
diff --git a/BlitzDiffApp/types.ts b/BlitzDiffApp/types.ts
index 64bc8a6dc..82209331c 100644
--- a/BlitzDiffApp/types.ts
+++ b/BlitzDiffApp/types.ts
@@ -1,16 +1,18 @@
-import { DefaultCtx, SessionContext } from "blitz"
-import { simpleRolesIsAuthorized } from "@blitzjs/server"
+import { DefaultCtx, SessionContext, SimpleRolesIsAuthorized } from "blitz"
 import { User } from "db"
 
+// Note: You should switch to Postgres and then use a DB enum for role type
+export type Role = "ADMIN" | "USER"
+
 declare module "blitz" {
   export interface Ctx extends DefaultCtx {
     session: SessionContext
   }
   export interface Session {
-    isAuthorized: typeof simpleRolesIsAuthorized
+    isAuthorized: SimpleRolesIsAuthorized<Role>
     PublicData: {
       userId: User["id"]
-      roles: string[]
+      role: Role
     }
   }
 }
