diff --git a/BlitzDiffApp/README.md b/BlitzDiffApp/README.md
index e269be399..6d00b24a4 100644
--- a/BlitzDiffApp/README.md
+++ b/BlitzDiffApp/README.md
@@ -33,8 +33,6 @@ DATABASE_URL=postgresql://<YOUR_DB_USERNAME>@localhost:5432/BlitzDiffApp_test
 Runs your tests using Jest.
 
 ```
-blitz test
-or
 yarn test
 ```
 
diff --git a/BlitzDiffApp/blitz.config.js b/BlitzDiffApp/blitz.config.js
index fa3795200..ce6217701 100644
--- a/BlitzDiffApp/blitz.config.js
+++ b/BlitzDiffApp/blitz.config.js
@@ -1,4 +1,4 @@
-const { sessionMiddleware, simpleRolesIsAuthorized } = require("@blitzjs/server")
+const { sessionMiddleware, simpleRolesIsAuthorized } = require("blitz")
 
 module.exports = {
   middleware: [
diff --git a/BlitzDiffApp/db/migrations/20210306000932_initial_migration/migration.sql b/BlitzDiffApp/db/migrations/20210306003910_initial_migration/migration.sql
similarity index 100%
rename from BlitzDiffApp/db/migrations/20210306000932_initial_migration/migration.sql
rename to BlitzDiffApp/db/migrations/20210306003910_initial_migration/migration.sql
diff --git a/BlitzDiffApp/package.json b/BlitzDiffApp/package.json
index abd00d886..2bf16e494 100644
--- a/BlitzDiffApp/package.json
+++ b/BlitzDiffApp/package.json
@@ -35,7 +35,7 @@
     "react": "0.0.0-experimental-3310209d0",
     "react-dom": "0.0.0-experimental-3310209d0",
     "react-error-boundary": "3.1.1",
-    "typescript": "4.2.3",
+    "typescript": "~4.1",
     "zod": "1.11.11",
     "final-form": "4.20.2",
     "react-final-form": "6.5.2"
diff --git a/BlitzDiffApp/types.ts b/BlitzDiffApp/types.ts
index cd9e61011..82209331c 100644
--- a/BlitzDiffApp/types.ts
+++ b/BlitzDiffApp/types.ts
@@ -1,5 +1,4 @@
-import { DefaultCtx, SessionContext } from "blitz"
-import { SimpleRolesIsAuthorized } from "@blitzjs/server"
+import { DefaultCtx, SessionContext, SimpleRolesIsAuthorized } from "blitz"
 import { User } from "db"
 
 // Note: You should switch to Postgres and then use a DB enum for role type
