diff --git a/BlitzDiffApp/app/auth/mutations/resetPassword.test.ts b/BlitzDiffApp/app/auth/mutations/resetPassword.test.ts
index d613716f5..0e0d44d94 100644
--- a/BlitzDiffApp/app/auth/mutations/resetPassword.test.ts
+++ b/BlitzDiffApp/app/auth/mutations/resetPassword.test.ts
@@ -8,7 +8,7 @@ beforeEach(async () => {
 
 const mockCtx: any = {
   session: {
-    $create: jest.fn,
+    create: jest.fn,
   },
 }
 
diff --git a/BlitzDiffApp/db/migrations/20210306011928_initial_migration/migration.sql b/BlitzDiffApp/db/migrations/20210306004559_initial_migration/migration.sql
similarity index 100%
rename from BlitzDiffApp/db/migrations/20210306011928_initial_migration/migration.sql
rename to BlitzDiffApp/db/migrations/20210306004559_initial_migration/migration.sql
diff --git a/BlitzDiffApp/types.ts b/BlitzDiffApp/types.ts
index 64bc8a6dc..1e460fbee 100644
--- a/BlitzDiffApp/types.ts
+++ b/BlitzDiffApp/types.ts
@@ -8,7 +8,7 @@ declare module "blitz" {
   }
   export interface Session {
     isAuthorized: typeof simpleRolesIsAuthorized
-    PublicData: {
+    publicData: {
       userId: User["id"]
       roles: string[]
     }
